pkgs/development/libraries/bashup-events/default.nix

```diff
@@ -0,0 +1,106 @@
+{ stdenv, lib, callPackage, runCommand, resholvePackage, fetchFromGitHub, bash, shellcheck, doCheck ? true, doInstallCheck ? true }:
+let
+  # extracting this so that it's trivial to test in other shells
+  installCheck = shell:
+    ''
+      echo "testing bashup.events in ${shell}"
+      ${shell} <<'EOF'
+      source $out/bin/bashup.events
+      neat(){
+        echo $0: Hi from event \'test event\'. I can have both $1 and $2 arguments.
+        exit 0
+      }
+      event on "test event" @2 neat curried
+      echo event registered
+      event emit "test event" runtime
+      exit 1 # fail if emitting event didn't exit clean
+      EOF
+    '';
+  common = { variant, version, branch, rev, hash, fake ? false, keep }: resholvePackage rec {
+    # bashup.events doesn't version yet but it has two variants with
+    # differing features/performance characteristics:
+    # - branch master: a variant for bash 3.2+
+    # - branch bash44: a variant for bash 4.4+
+    pname = "bashup-events${variant}-unstable";
```

Comment:
The unstable belongs to the version in this case.

Reply:
done in c4043d586dff8fd52c47070908607e7ee0f82a0f